packages/react/src/hooks/useCounts.ts

```diff
@@ -68,9 +68,11 @@ export const useCounts = (props: UseCountsProps): UseCountsResult => {
 
       // eslint-disable-next-line no-plusplus
       for (let i = 0; i < existingCounts.length; i++) {
-        const countReceived = countsReceived.find((c) => areTagsEqual(c.filter.tags, existingCounts[i]?.filter.tags));
-
-        newCounts.push(countReceived || oldCounts![i]);
+        const countReceived = countsReceived.find((c) => areTagsEqual(c.filter.tags, existingCounts[i].filter.tags));
```

Comment:
When getting counts from the API we look for the single count by comparing its tags.
The issue was that the `existingCounts` were filled with `undefined,` resulting in the failure to compare the tags.